akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -343,9 +343,9 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   private val largeEnvelopeBufferPool = new EnvelopeBufferPool(settings.Advanced.MaximumLargeFrameSize, settings.Advanced.MaximumPooledBuffers)
 
   private val inboundEnvelopePool = ReusableInboundEnvelope.createObjectPool(capacity = 16)
-  // FIXME capacity of outboundEnvelopePool should probably be derived from the sendQueue capacity
-  //       times a factor (for reasonable number of outbound streams)
-  private val outboundEnvelopePool = ReusableOutboundEnvelope.createObjectPool(capacity = 3072 * 2)
+  // The outboundEnvelopePool is shared among all outbound associations
+  private val outboundEnvelopePool = ReusableOutboundEnvelope.createObjectPool(capacity =
+    settings.Advanced.OutboundMessageQueueSize * 3)
```

Comment:
yes, we could multiply with number of lanes, but on the other hand it might not be needed to use a too big pool and a sometimes a too small pool will not hurt too much since it's plain objects